### PR TITLE
Updated broken link for Yul optimizations

### LINKED ISSUE #13025: Broken link for Yul optimizations steps
## Page
https://github.com/ethereum/solidity/blob/develop/docs/internals/optimizer.rst

## Abstract
This link is dead.
> Available abbreviations are listed in the [Yul optimizer docs](https://github.com/ethereum/solidity/blob/develop/docs/internals/yul.rst#optimization-step-sequence).
## Pull request

If someone can point indicate where this should link, I'll submit a PR. I don't see the "optimization step sequence" heading on this page either https://github.com/ethereum/solidity/blob/develop/docs/yul.rst

### PR DESCRIPTION
**Pull request by PrajwalBorkar**
Fixes #13025

### CODE REVIEW

File: docs/internals/optimizer.rst [hunk 1]
```diff
@@ -330,7 +330,7 @@ the ``--yul-optimizations`` option:
 The sequence inside ``[...]`` will be applied multiple times in a loop until the Yul code
 remains unchanged or until the maximum number of rounds (currently 12) has been reached.
 
-Available abbreviations are listed in the `Yul optimizer docs <yul.rst#optimization-step-sequence>`_.
+Available abbreviations are listed in the `Yul optimizer docs <https://docs.soliditylang.org/en/v0.8.13/yul.html#optimization-step-sequence>`_.
```
Comment (cameel) [inline docs/internals/optimizer.rst]: This must be a reference, not a hard-coded link. Please use a proper reference. See [Cross-referencing with Sphinx > Explicit targets](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#explicit-targets)

**Review by cameel — CHANGES_REQUESTED** [review]

File: docs/internals/optimizer.rst (hunk 1 above)
Comment (PrajwalBorkar) [reply docs/internals/optimizer.rst]: According to the doc you shared, the previous link looks correct to me. And I don't know properly how to reference other than the original one.

Comment (axic) [reply docs/internals/optimizer.rst]: Why does this link has to refer `yul.rst` in the first place? Should just refer to a section.

Comment (PrajwalBorkar) [reply docs/internals/optimizer.rst]: > Why does this link has to refer `yul.rst` in the first place? Should just refer to a section.

If you know the correct reference, please paste here so I can update😄

Comment (cameel) [reply docs/internals/optimizer.rst]: Link updated.

**Review by cameel — APPROVED** [review]

**Pull request merged**